src/iterative.F90

```diff
@@ -3270,54 +3287,70 @@ subroutine calculate_transmissions(H,S,Ec,SelfEneR,ni,nf,str,tun_proj,tun_mat)
       ibl = str%cblk(i)
       ESH(ibl,ibl)%val = ESH(ibl,ibl)%val-SelfEneR(i)%val
     end do
-    call allocate_gsm(gsmr,nbl)
 
-    !Iterative calculation up with gsmr
-    call calculate_gsmr_blocks(ESH,nbl,2)
+    ! Fall here when there are 2 contacts for fast transmission
+    if (ncont == 2 .and. size(ni) == 1) then
+      nit=ni(1)
+      nft=nf(1)
+      ! order blocks in such a way that nt is that with smaller index
+      if (str%cblk(nit).lt.str%cblk(nft)) then
+        nt = str%cblk(nit)
+      else
+        nt = str%cblk(nft)
+      endif
+      call allocate_gsm(gsmr,nbl)
+      call calculate_gsmr_blocks(ESH,nbl,2,keepall)
+      call allocate_blk_dns(Gr,nbl)
+      call calculate_Gr_tridiag_blocks(ESH,1)
+      ! This is needed if the lowest block is not 1
+      if (nt.gt.1) then
+        call calculate_Gr_tridiag_blocks(ESH,2,nt)
```

Comment:
Don't we need also the gsmr which are destroyed with the option `keepall=.false.` in this case?
Do we actually really support this case, that the leftmost contact is interacting with a PL other than the first one?

Reply:
In principle it can work if the leftmost contact is not on the first PL, but yes in this case we need the gr removed before. I think this is a special case that should go on the general case. Ok we should also check the lowest contact is the first to fall in this fast thing.

Reply:
fixed in next commit, resolving